crates/sui-indexer-alt-framework/src/pipeline/concurrent/commit_watermark.rs

```diff
@@ -255,3 +255,303 @@ pub(super) fn commit_watermark<H: Handler + 'static>(
+        ) -> anyhow::Result<usize> {
+            Ok(0)
+        }
+    }
+
+    async fn test_db() -> (TempDb, Db) {
+        let temp_db = TempDb::new().unwrap();
+        let db_config = DbArgs::new(temp_db.database().url().clone());
+        let db = Db::new(db_config).await.unwrap();
+        db.run_migrations(MIGRATIONS).await.unwrap();
+        (temp_db, db)
+    }
+
+    // Create a new commit watermark with the given checkpoint.
+    // For the testing done here, only checkpoint matters so all other fields are arbitrary.
+    async fn new_commit_watermark(checkpoint: i64) -> CommitterWatermark<'static> {
+        CommitterWatermark {
+            pipeline: "test".into(),
+            epoch_hi_inclusive: 1,
+            checkpoint_hi_inclusive: checkpoint,
+            tx_hi: 10,
+            timestamp_ms_hi_inclusive: 100,
+        }
+    }
+
+    #[tokio::test]
+    async fn test_commit_watermark_basic() {
+        let (_temp_db, db) = test_db().await;
+        let registry = Registry::new_custom(Some("indexer_alt".to_string()), None).unwrap();
+        let metrics = Arc::new(IndexerMetrics::new(&registry));
+        let cancel = CancellationToken::new();
+        let (tx, rx) = mpsc::channel(100);
+
+        let initial_watermark = new_commit_watermark(5).await;
+
+        let handle = commit_watermark::<TestHandler>(
+            Some(initial_watermark),
+            CommitterConfig::default(),
+            false,
+            rx,
+            db.clone(),
+            metrics,
+            cancel.clone(),
+        );
+
+        // Send some watermark parts
+        let part = WatermarkPart {
+            watermark: new_commit_watermark(6).await,
+            batch_rows: 0,
+            total_rows: 0,
+        };
+        tx.send(vec![part]).await.unwrap();
+
+        // Let the task process the watermark
+        tokio::time::sleep(Duration::from_millis(500)).await;
+
+        let watermark =
+            CommitterWatermark::read_highest_watermark(&mut db.connect().await.unwrap(), "test")
+                .await
+                .unwrap()
+                .unwrap();
+        assert_eq!(watermark.checkpoint_hi_inclusive, 6);
+
+        // Cancel and wait for shutdown
+        cancel.cancel();
+        handle.await.unwrap();
+    }
+
+    #[tokio::test]
+    async fn test_commit_watermark_skip() {
+        let (_temp_db, db) = test_db().await;
+        let registry = Registry::new_custom(Some("indexer_alt".to_string()), None).unwrap();
+        let metrics = Arc::new(IndexerMetrics::new(&registry));
+        let cancel = CancellationToken::new();
+        let (_tx, rx) = mpsc::channel(100);
+
+        let handle = commit_watermark::<TestHandler>(
+            None,
+            CommitterConfig::default(),
+            true,
+            rx,
+            db,
+            metrics,
+            cancel,
+        );
+
+        // Task should exit immediately when skip_watermark is true
+        handle.await.unwrap();
```

Comment:
Is there a way for you to test that nothing is written to the DB or something? This test doesn't really tell me that the task exited immediately.